packages/cli/src/commands/info/action.ts

```diff
@@ -0,0 +1,185 @@
+  const targetGraph = createTargetGraph({
+    logger,
+    root,
+    dependencies: options.dependencies,
+    dependents: options.dependents && !options.to, // --to is a short hand for --scope + --no-dependents
+    ignore: options.ignore.concat(config.ignore),
+    pipeline: config.pipeline,
+    repoWideChanges: config.repoWideChanges,
+    scope: (options.scope ?? []).concat(options.to ?? []), // --to is a short hand for --scope + --no-dependents
+    since: options.since,
+    outputs: config.cacheOptions.outputGlob,
+    tasks,
+    packageInfos,
+  });
+
+  return targetGraph;
+}
+
+function prepareAndGetFilteredPackages(config, logger, root, options, packageInfos) {
+  const scope = getFilteredPackages({
+    root,
+    packageInfos,
+    logger,
+    includeDependencies: options.dependencies,
+    includeDependents: options.dependents && !options.to, // --to is a short hand for --scope + --no-dependents
+    since: options.since,
+    scope: (options.scope ?? []).concat(options.to ?? []), // --to is a short hand for --scope + --no-dependents
+    repoWideChanges: config.repoWideChanges,
+    sinceIgnoreGlobs: options.ignore.concat(config.ignore),
+  });
+
+  return scope;
+}
+
+function initialOutputGenerator(command, scope) {
+  const output: Output = {
+    msg: "info",
+    data: {
+      command: command.args,
+      scope: scope,
+      packageTasks: [],
+    },
+  };
+  return output;
+}
+
+function processTargets(targets, packageInfos, config, output) {
+  const packageTasksByPackage: PackageTasksByPackage = {};
+
+  for (const target of targets.values()) {
+    if (shouldSkipTarget(target, packageInfos)) {
+      continue;
+    }
+
+    const startIdIndex = target.dependencies.indexOf(getStartTargetId());
+    target.dependencies.splice(startIdIndex, 1);
+
+    const packageTask = generatePackageTask(target, config);
+    packageTasksByPackage[packageTask.package] = packageTasksByPackage[packageTask.package] || [];
+    packageTasksByPackage[packageTask.package].push(packageTask);
+  }
+
+  appendPackageTasksToOutput(packageTasksByPackage, output);
+
+  return output;
+}
+
+function shouldSkipTarget(target, packageInfos) {
+  return target.id === getStartTargetId() || !packageInfos[target.packageName ?? ""]?.scripts?.[target.task];
+}
+
+function appendPackageTasksToOutput(packageTasksByPackage: PackageTasksByPackage, output: Output) {
+  for (const [, packageTasks] of Object.entries(packageTasksByPackage)) {
+    output.data.packageTasks.push(...packageTasks);
+  }
+}
+
+function generatePackageTask(target, config): PackageTask {
+  const command = generateCommand(target, config);
+  const workingDirectory = getWorkingDirectory(target);
+
+  const packageTask: PackageTask = {
+    id: target.id,
+    command,
+    dependencies: target.dependencies,
+    workingDirectory,
+    package: target.packageName,
+    task: target.task,
+  };
+
+  return packageTask;
+}
+
+function generateCommand(target, config) {
```

Comment:
@kenotron Maybe you can give me some pointers on this.
Wasn't really sure on how to implement this.

Reply:
this looks alright... it's not perfect, but we'll start here.